### PR TITLE
DAML-LF. fix computation of cid

### LINKED ISSUE #5485: Contract id computation scheme is buggy
When computing the seed for the child node, the current implementation uses thetotal number of nodes in the transaction instead of the current children of node in the current node.

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
This fixes #5485 

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by remyhaemmerle-da** [comment]
Will add some tests

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala [hunk 1]
```diff
@@ -146,21 +146,19 @@ final class Engine {
...
       submitters: Set[Party],
       nodes: Seq[GenNode.WithTxValue[Value.NodeId, Value.ContractId]],
       ledgerEffectiveTime: Time.Timestamp,
   ): Result[(Transaction.Transaction, Boolean)] = {
 
-    val transactionSeedAndSubmissionTime = submissionSeedAndTime.map {
-      case (seed, time) =>
-        crypto.Hash.deriveTransactionSeed(seed, participantId, time) -> time
-    }
+    remy.log(" *** ")
```
Comment (andreaslochbihler-da) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala:155]: That looks like a left-over.

File: daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala [hunk 2]
```diff
@@ -1417,6 +1427,69 @@ class EngineTest extends WordSpec with Matchers with EitherValues with BazelRunf
...
+        engine
+          .validate(tx, let, participant, Some(submissionSeed -> metaData.submissionTime))
+          .consume(_ => None, lookupPackage, _ => None)
+
+      run(0).flatMap { case (tx, metaData) => validate(tx, metaData) } shouldBe Right(())
+      run(3).flatMap { case (tx, metaData) => validate(tx, metaData) } shouldBe Right(())
+    }
+
+    "be partially reinterpretable" in {
+      val Right((tx, metaData)) = run(3)
```
Comment (remyhaemmerle-da) [inline daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala:1475]: This test fails without the fix to `PartialTransaction`

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala [hunk 3]
```diff
@@ -453,6 +453,6 @@ case class PartialTransaction(
   }
 
   def deriveChildSeed: Option[crypto.Hash] =
-    context.contextSeed.map(crypto.Hash.deriveNodeSeed(_, nodes.size))
+    context.contextSeed.map(crypto.Hash.deriveNodeSeed(_, context.children.length))
```
Comment (remyhaemmerle-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala:457]: This is the actual fix

**Pull request merged**